java/java-analysis-impl/src/com/intellij/codeInspection/dataFlow/Mutability.java

```diff
@@ -64,6 +64,8 @@
   public static final @NotNull String UNMODIFIABLE_VIEW_ANNOTATION = UNMODIFIABLE_VIEW.myAnnotation;
   private static final @NotNull CallMatcher STREAM_COLLECT = CallMatcher.instanceCall(
     CommonClassNames.JAVA_UTIL_STREAM_STREAM, "collect").parameterTypes("java.util.stream.Collector");
+  private static final @NotNull CallMatcher STREAM_COLLECT_TO_LIST = CallMatcher.instanceCall(
+    CommonClassNames.JAVA_UTIL_STREAM_STREAM, "toList");
```

Comment:
I would add `.withLanguageLevelAtLeast(JDK_16)`. This matcher can match `toList()` method declared in inheritors, even if `Stream.toList()` is absent. As before JDK_16 there were no `toList()` in the Stream, inheritors were free to assign any other semantics to the result of `toList()`, including the modifiable list. This was true for older versions of `StreamEx` library and probably true for `jOOL` library (haven't checked). So if the project was not updated to Java 16 and uses one of these libraries, the libraries don't violate the contract there but we will have incorrect warnings.

Reply:
Thank you for your explanation. Fixed it